src/main/java/Model/FEN.java

```diff
@@ -0,0 +1,187 @@
+package Model;
+
+
+import Model.pieces.Piece;
+
+import static java.util.Objects.isNull;
+
+public class FEN {
```

Comment:
this FEN class is awesome.

its clearly got complicated logic and very hard to eyeball... perfect candidate to be unit tested..!

Reply:
Once merged will create another Branch for Testing